tests/README.rst

```diff
@@ -0,0 +1,50 @@
+Functional Testing
+==================
+The directory structure, files, and tests found in this directory all work
+together to provide:
+
+* a set of machines (or even a single one) so that ceph-ansible can run against
+* a "scenario" configuration file in Python, that defines what nodes are
+  configured to what roles and what 'components' they will test
+* tests (in functional/tests/) that will all run unless skipped explicitly when
+  testing a distinct feature dependant on the ansible run.
+
+
+Example run
+-----------
+The following is the easiest way to try this out locally. Both Vagrant and
+VirtualBox are required. Ensure that ``py.test`` and ``pytest-xdist`` are
+installed (with pip on a virtualenv) by using the ``requirements.txt`` file in
+the ``tests`` directory::
+
+    pip install -r requirements.txt
+
+Choose a directory in ``tests/functional`` that has 3 files:
+
+* ``Vagrantfile``
+* ``vagrant_variables.yml``
+* A Python ("scenario") file.
+
+For example in: ``tests/functional/ubuntu/16.04/mon/initial_members``::
+
+    tree .
+    .
+    ├── Vagrantfile -> ../../../../../../Vagrantfile
+    ├── scenario.py
+    └── vagrant_variables.yml
```

Comment:
This PR doesn't have a ``vagrant_variables.yml`` in that directory. Are we expecting users to put their own ``vagrant_variables.yml`` in these directories? I was thinking each scenario would have one pre-defined as the config there will need to match the scenario being tested.